2022-10-11 03:35:21,161 INFO    Thread-8  :18213 [wandb_setup.py:_flush():75] Loading settings from /root/.config/wandb/settings
2022-10-11 03:35:21,162 INFO    Thread-8  :18213 [wandb_setup.py:_flush():75] Loading settings from wandb/settings
2022-10-11 03:35:21,162 INFO    Thread-8  :18213 [wandb_setup.py:_flush():75] Loading settings from environment variables: {'project': 'meme_experiments', 'entity': 'vietnguyen', 'root_dir': '/content/SSLMemes', 'sweep_id': 'x8t672gd', 'run_id': 'rzeks0ng', 'sweep_param_path': '/content/SSLMemes/wandb/sweep-x8t672gd/config-rzeks0ng.yaml'}
2022-10-11 03:35:21,162 INFO    Thread-8  :18213 [wandb_setup.py:_flush():75] Inferring run settings from compute environment: {'program_relpath': 'src/train.py', 'program': '/content/SSLMemes/src/train.py'}
2022-10-11 03:35:21,162 INFO    Thread-8  :18213 [wandb_init.py:_log_setup():405] Logging user logs to /content/SSLMemes/wandb/run-20221011_033521-rzeks0ng/logs/debug.log
2022-10-11 03:35:21,163 INFO    Thread-8  :18213 [wandb_init.py:_log_setup():406] Logging internal logs to /content/SSLMemes/wandb/run-20221011_033521-rzeks0ng/logs/debug-internal.log
2022-10-11 03:35:21,163 INFO    Thread-8  :18213 [wandb_init.py:init():439] calling init triggers
2022-10-11 03:35:21,163 INFO    Thread-8  :18213 [wandb_init.py:init():443] wandb.init called with sweep_config: {'batch_size': 32, 'dropout': 0, 'epochs': 1, 'hidden_size': 128, 'learning_rate': 0.044071303807555, 'optimizer': 'sgd'}
config: {}
2022-10-11 03:35:21,163 INFO    Thread-8  :18213 [wandb_init.py:init():492] starting backend
2022-10-11 03:35:21,164 INFO    Thread-8  :18213 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=fork,spawn,forkserver, using: spawn
2022-10-11 03:35:21,200 INFO    Thread-8  :18213 [backend.py:ensure_launched():219] starting backend process...
2022-10-11 03:35:21,229 INFO    Thread-8  :18213 [backend.py:ensure_launched():225] started backend process with pid: 18377
2022-10-11 03:35:21,247 INFO    Thread-8  :18213 [wandb_init.py:init():501] backend started and connected
2022-10-11 03:35:21,276 INFO    Thread-8  :18213 [wandb_run.py:_config_callback():992] config_cb None None {'batch_size': 32, 'dropout': 0, 'epochs': 1, 'hidden_size': 128, 'learning_rate': 0.044071303807555, 'optimizer': 'sgd'}
2022-10-11 03:35:21,289 INFO    Thread-8  :18213 [wandb_init.py:init():565] updated telemetry
2022-10-11 03:35:21,413 INFO    Thread-8  :18213 [wandb_init.py:init():596] communicating run to backend with 30 second timeout
2022-10-11 03:35:28,983 INFO    Thread-8  :18213 [wandb_run.py:_on_init():1759] communicating current version
2022-10-11 03:35:29,056 INFO    Thread-8  :18213 [wandb_run.py:_on_init():1763] got version response upgrade_message: "wandb version 0.13.4 is available!  To upgrade, please run:\n $ pip install wandb --upgrade"

2022-10-11 03:35:29,056 INFO    Thread-8  :18213 [wandb_init.py:init():625] starting run threads in backend
2022-10-11 03:35:33,981 INFO    Thread-8  :18213 [wandb_run.py:_console_start():1733] atexit reg
2022-10-11 03:35:33,981 INFO    Thread-8  :18213 [wandb_run.py:_redirect():1606] redirect: SettingsConsole.REDIRECT
2022-10-11 03:35:33,984 INFO    Thread-8  :18213 [wandb_run.py:_redirect():1611] Redirecting console.
2022-10-11 03:35:33,990 ERROR   Thread-8  :18213 [wandb_run.py:_redirect():1670] Failed to redirect.
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/wandb_run.py", line 1663, in _redirect
    out_redir.install()
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/lib/redirect.py", line 709, in install
    _WSCH.add_fd(self._pipe_read_fd)
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/lib/redirect.py", line 659, in add_fd
    self._register()
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/lib/redirect.py", line 644, in _register
    old_handler = signal.signal(signal.SIGWINCH, lambda *_: None)
  File "/usr/lib/python3.7/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread
2022-10-11 03:35:33,992 INFO    Thread-8  :18213 [wandb_init.py:init():664] run started, returning control to user process
2022-10-11 03:35:33,994 INFO    Thread-8  :18213 [wandb_run.py:_config_callback():992] config_cb None None {'img_feature_path': 'data/features/visualbert/', 'train_csv_path': 'data/splits/random/memotion_train.csv', 'val_csv_path': 'data/splits/random/memotion_val.csv', 'model_type': 'visualbert', 'model_path': 'uclanlp/visualbert-nlvr2-coco-pre', 'epoch': 100, 'eval_step': 100, 'amp': True, 'output_dir': './tmp', 'checkpoint_step': 100, 'random_seed': 42, 'resume_training': False, 'semi_supervised': False}
2022-10-11 03:35:34,371 INFO    Thread-8  :18213 [wandb_run.py:_config_callback():992] config_cb None None {'output_hidden': 768}
